lib/Raven/Client.php

```diff
@@ -747,32 +735,10 @@ public function send(&$data)
             return;
         }
 
-        $message = $this->encode($data);
-
-        $headers = [
-            'User-Agent' => static::getUserAgent(),
-            'X-Sentry-Auth' => $this->getAuthHeader(),
-            'Content-Type' => 'application/octet-stream'
-        ];
-
-        $config = $this->getConfig();
-        $server = sprintf('%s/api/%d/store/', $config->getServer(), $config->getProjectId());
-
-        $this->send_remote($server, $message, $headers);
-    }
+        $uri = sprintf('api/%d/store/', $this->getConfig()->getProjectId());
+        $request = $this->requestFactory->createRequest('POST', $uri, [], JSON::encode($data));
 
-    /**
-     * Send data to Sentry
-     *
-     * @param string       $url     Full URL to Sentry
-     * @param array|string $data    Associative array of data to log
-     * @param array        $headers Associative array of headers
-     */
-    protected function send_remote($url, $data, $headers = [])
-    {
-        $parts = parse_url($url);
-        $parts['netloc'] = $parts['host'].(isset($parts['port']) ? ':'.$parts['port'] : null);
-        $this->send_http($url, $data, $headers);
+        $this->httpClient->sendAsyncRequest($request)->wait(true);
```

Comment:
>We are also going to need to support a 429 backoff response from the Sentry server.

At the moment there is no code neither in `1.x` nor in `2.x` that handles failure of sent events, as far as I know. We should indeed implement it, but maybe it's better to finish the implementation of Httplug in this PR and think about that topic in a new one

>Going async allows us to have a callback right?

Yes, this is exact. Right now I append the Httplug plugin that takes any error response and converts it into an exception, but I'm not sure this is a good idea as probably the exception handler would catch it and try to resent the event, starting an infinite loop (maybe?) in case 429 is the error code of the response